bokehjs/src/lib/models/annotations/scale_bar.ts

```diff
@@ -0,0 +1,627 @@
+import {Annotation, AnnotationView} from "./annotation"
+import {Dimensional, MetricLength} from "./dimensional"
+import {Range} from "../ranges/range"
+import {Range1d} from "../ranges/range1d"
+import {Align, Anchor, Orientation, Location} from "core/enums"
+import type * as visuals from "core/visuals"
+import * as mixins from "core/property_mixins"
+import type * as p from "core/properties"
+import {TextBox} from "core/graphics"
+import type {Size} from "core/layout"
+import {SideLayout, Panel} from "core/layout/side_panel"
+import {BBox} from "core/util/bbox"
+import type {Context2d} from "core/util/canvas"
+import type {Layoutable} from "core/layout"
+import {Column, Row, ContentLayoutable, Sizeable} from "core/layout"
+import type {ContinuousAxis, ContinuousAxisView} from "../axes/continuous_axis"
+import {LinearAxis} from "../axes/linear_axis"
+import {Ticker} from "../tickers/ticker"
+import {FixedTicker} from "../tickers/fixed_ticker"
+import type {Scale} from "../scales/scale"
+import {LinearScale} from "../scales/linear_scale"
+import {CoordinateTransform} from "../coordinates/coordinate_mapping"
+import {build_view} from "core/build_views"
+import {clamp} from "core/util/math"
+import {process_placeholders, sprintf} from "core/util/templating"
+import {Enum} from "../../core/kinds"
+
+const {round} = Math
+
+const LengthSizing = Enum("adaptive", "exact")
+type LengthSizing = typeof LengthSizing["__type__"]
+
+class TextLayout extends ContentLayoutable {
+
+  constructor(readonly text: TextBox) {
+    super()
+  }
+
+  _content_size(): Sizeable {
+    return new Sizeable(this.text.size())
+  }
+}
+
+/*
+class FixedLayout extends ContentLayoutable {
```

Comment:
Is this here for future improvement, or can it just be removed now?

Reply:
The first design used a custom bar layout. The second a mixture of bar layout and an axis. Now it's all implemented using an axis. Thus all the commented out code isn't needed anymore. Though I will keep  this one under `core/layout`, as it may be useful at some point.